ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/eth1/Eth1Address.java

```diff
@@ -27,17 +31,77 @@ public class Eth1Address extends Bytes20 {
           .format("byte")
           .build();
 
-  public static final Eth1Address ZERO = new Eth1Address(Bytes.wrap(new byte[SIZE]));
+  public static final Eth1Address ZERO =
+      new Eth1Address("0x0000000000000000000000000000000000000000");
+
+  private final String encodedAddress;
+
+  private Eth1Address(String value) {
+    super(Bytes.fromHexString(value));
+    if (!value.startsWith("0x")) {
+      value = "0x" + value;
+    }
+    this.encodedAddress = toChecksumAddress(value);
+    if (isMixedCase(value)) {
+      checkArgument(
+          value.equals(encodedAddress),
+          "Eth1Address fails checksum:\n got: %s\n exp: %s",
+          value,
+          encodedAddress);
+    }
+  }
 
-  public Eth1Address(final Bytes bytes) {
-    super(bytes);
+  private static boolean isMixedCase(String value) {
+    String valueWithoutPrefix = value.replace("0x", "");
+    return !valueWithoutPrefix.toLowerCase().equals(valueWithoutPrefix)
+        && !valueWithoutPrefix.toUpperCase().equals(valueWithoutPrefix);
   }
```

Comment:
We can use `StringUtils.isMixedCase` (from commons lang3)

Reply:
Ah perfect. I didn't know that existed. Thank you.